￼ fix(app): minimize command page length when only retrieving run total length

b-cooper opened the pull request:
# Overview

Instead of fetching an entire window of commands from each run on the historic run details page to
prep the download run log button, fetch zero commands in order to retrieve the total run command
count.

re [RAUT-315](https://opentrons.atlassian.net/browse/RAUT-315)

# Test Plan

- confirm that downloading the run data from the overflow menu on the historic runs list still functions as expected.



[RAUT-315]: https://opentrons.atlassian.net/browse/RAUT-315?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

SyntaxColoring commented on app/src/organisms/Devices/HistoricalProtocolRunOverflowMenu.tsx:
```diff
@@ -130,15 +104,22 @@ function MenuDropdown(props: MenuDropdownProps): JSX.Element {
+
+  const commands = useAllCommandsQuery(
+    runId,
+    { cursor: 0, pageLength: RUN_LOG_WINDOW_SIZE },
```
Ideas for the future:

* We should see about making this `0` or something, if all you need is the `totalLength`. Or maybe we can offer the `totalLength` in a different endpoint.
* Or, better yet, we should see about giving you a way to download all commands without knowing the length. Like `pageLength="toEnd"` or something.

SyntaxColoring reviewed (approved):
> confirm that landing on the device detail page doesn't lead to GET requests to any /runs/:runId/commands requests.

Confirmed. Thank you!

brenthagen commented on app/src/organisms/Devices/HistoricalProtocolRunOverflowMenu.tsx:
```diff
@@ -192,6 +173,14 @@ function MenuDropdown(props: MenuDropdownProps): JSX.Element {
         {t('delete_run')}
       </MenuItem>
+      {runTotalCommandCount != null && showDownloadRunLogToast ? (
+        <DownloadRunLogToast
```
this "toast" component calls the download function - moving it to the overflow menu unmounts the toast component on click and (at least for me) doesn't trigger the download.

one quick fix: remove the `closeOverflowMenu(e)` call from line 102 of `onDownloadClick`, which leaves the overflow open during the download for the user to close later. maybe not ideal.

`DownloadRunLogToast` was refactored in the toaster oven PR into edge, but it still has the same problem. probably an asynchronous download approach like `electron-dl` would be the way to go eventually: https://github.com/Opentrons/opentrons/pull/11797/files#r1039981096

SyntaxColoring commented:
If this feels at all nontrivial/dicey/more-nuanced-than-we-thought, let's drop it from `hotfix_6.2.1`.

b-cooper committed:
fix(app): minimize command page length when only retrieving run total length

Instead of fetching an entire window of commands from each run on the historic run details page to
prep the download run log button, fetch zero commands in order to retrieve the total run command
count.

re RAUT-315

SyntaxColoring reviewed (approved):
The code of this more conservative rewrite looks correct to me. I've smoke-tested this in-person with @b-cooper and it seems to make a meaningful improvement—400 ms per request instead of several seconds per request, with whatever protocols the office OT-2 happens to have stored.

b-cooper commented:
Good catches @brenthagen  and @SyntaxColoring, I've updated the PR to bite off a little less. Now the only change from the existing approach is that the page length of the `commands` fetch is 0 instead of 60.  We only need the `totalLength` from this response anyway, and this leads to a performance improvement short of the larger refactor we'd like to do here soon.

b-cooper committed:
lint

brenthagen reviewed (approved):
makes sense to me, checked that download run log still downloads ✅
